tests/Unit/Helpers/NumericalAlgorithms/Interpolation/InterpolateOnElementTestHelpers.hpp

```diff
@@ -0,0 +1,243 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Domain/BlockLogicalCoordinates.hpp"
+#include "Domain/Creators/Shell.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementLogicalCoordinates.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/InitialElementIds.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Framework/ActionTesting.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/Interpolation/Actions/InterpolationTargetVarsFromElement.hpp"
+#include "NumericalAlgorithms/Interpolation/InterpolationTarget.hpp"
+#include "NumericalAlgorithms/Interpolation/PointInfoTag.hpp"
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "Time/Slab.hpp"
+#include "Time/Tags.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeStepId.hpp"
+
+/// Holds code that is shared between multiple tests. Currently used by
+/// - Test_InterpolateToTarget
+/// - Test_InterpolateWithoutInterpolatorComponent.
+namespace InterpolateOnElementTestHelpers {
+
+namespace Tags {
+// Simple Variables tag for test.
+struct TestSolution : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+// Tag holding inertial-frame target points for test.
+struct TestTargetPoints : db::SimpleTag {
+  using type = tnsr::I<DataVector, 3, Frame::Inertial>;
+};
+// ComputeItem for test.
+struct MultiplyByTwo : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+struct MultiplyByTwoComputeItem : MultiplyByTwo, db::ComputeTag {
+  static Scalar<DataVector> function(const Scalar<DataVector>& x) noexcept {
+    auto result = make_with_value<Scalar<DataVector>>(x, 0.0);
+    get<>(result) = get<>(x) * 2.0;
+    return result;
+  }
+  using argument_tags = tmpl::list<TestSolution>;
+  using base = MultiplyByTwo;
+};
+}  // namespace Tags
+
+template <typename TagName, typename VarsTagList>
+void fill_variables(
+    const gsl::not_null<Variables<VarsTagList>*> vars,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& coords) noexcept {
+  // Some analytic solution used to fill the volume data and
+  // to test the interpolated data.
+  auto& solution = get<TagName>(*vars);
+  get(solution) =
+      (2.0 * get<0>(coords) + 3.0 * get<1>(coords) + 5.0 * get<2>(coords));
+
+  // NOLINTNEXTLINE clang-tidy is unhappy with `if constexpr` braces.
```

Comment:
Is this still true with clang-tidy 10?

Reply:
Cool. clang-tidy 10 now works. Removing the NOLINTS...